### PR TITLE
python311Packages.natasha: init at 1.6.0

### PR DESCRIPTION
**Pull request by npatsakula**
## Description of changes

Initial version of [Natasha](https://github.com/natasha) project (`natasha` + dependencies) derivation.

1. Natasha: https://github.com/natasha/natasha
2. Yargy: https://github.com/natasha/yargy
3. Slovnet: https://github.com/natasha/slovnet
4. Razdel: https://github.com/natasha/razdel
5. Ipymarkup: https://github.com/natasha/ipymarkup

Some tests disabled due to sandbox unfriendly tests scripts, e.g. `slovnet` trying to download model binary artifacts.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/natasha/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ buildPythonPackage, fetchPypi, python, pymorphy2, razdel, navec, slovnet
+, yagry, ipymarkup, pytest, isPy3k, lib }:
```
Comment (natsukium) [inline pkgs/development/python-modules/natasha/default.nix:2]:
```suggestion
{ lib
, buildPythonPackage
, ... 
}:
```
Would you please add a line per item for future diff?

File: pkgs/development/python-modules/natasha/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+buildPythonPackage rec {
+  pname = "natasha";
+  version = "1.6.0";
+  format = "setuptools";
+
+  disabled = !isPy3k;
+
```
Comment (natsukium) [inline pkgs/development/python-modules/natasha/default.nix:9]:
```suggestion
```
We no longer maintain python2, so this is redundant.

File: pkgs/development/python-modules/natasha/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    hash = "sha256-Rgguazgq06a8B9jeRnfHD5VTR+Xrd+8OCsQUfaGLEq0=";
+  };
+
+  propagatedBuildInputs = [ pymorphy2 navec razdel slovnet yagry ipymarkup ];
+
+  nativeCheckInputs = [ pytest ];
+
+  checkPhase = ''
+    ${python.interpreter} -m pytest tests
+  '';
```
Comment (natsukium) [inline pkgs/development/python-modules/natasha/default.nix:21]: Please use `pytestCheckHook`.

File: pkgs/development/python-modules/natasha/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  pythonImportCheck = [ "natasha" ];
+
+  meta = with lib; {
+    description =
+      "Natasha solves basic NLP tasks for Russian language: tokenization, sentence segmentation, word embedding, morphology tagging, lemmatization, phrase normalization, syntax parsing, NER tagging, fact extraction";
```
Comment (natsukium) [inline pkgs/development/python-modules/natasha/default.nix:27]: This sentence is a bit long for description, IMO.
How about citing `Solves basic Russian NLP tasks, API for lower level Natasha projects` from GitHub or `Named-entity recognition for russian language`?
https://github.com/natasha/natasha/blob/ebe33ff1be333af0059aa34da825dfd4abb15dc2/setup.py#L13

File: pkgs/development/python-modules/natasha/ipymarkup.nix
Comment (natsukium) [inline pkgs/development/python-modules/natasha/ipymarkup.nix:1]: Could you please move this file to `pkgs/development/python-modules/ipymarkup/default.nix`?

**Review by natsukium — CHANGES_REQUESTED** [review]
Please split your commit into separate commits, `python311Packages.natasha: init at 1.6.0`, `python311Packages.ipymarkup: init at 0.9.0`, and so on.

To avoid similar suggestions, I left only a few comments.
Please apply the same to other files.

File: pkgs/development/python-modules/natasha/default.nix (hunk 1 above)
Comment (npatsakula) [reply pkgs/development/python-modules/natasha/default.nix]: Done!

File: pkgs/development/python-modules/natasha/default.nix (hunk 3 above)
Comment (npatsakula) [reply pkgs/development/python-modules/natasha/default.nix]: Yes, thank you for mention it. Done!

File: pkgs/development/python-modules/natasha/ipymarkup.nix
Comment (npatsakula) [reply pkgs/development/python-modules/natasha/ipymarkup.nix:1]: Yes, other dependencies moved as well.

File: pkgs/development/python-modules/natasha/default.nix (hunk 2 above)
Comment (npatsakula) [reply pkgs/development/python-modules/natasha/default.nix]: Done!

File: pkgs/development/python-modules/natasha/default.nix (hunk 4 above)
Comment (npatsakula) [reply pkgs/development/python-modules/natasha/default.nix]: I shortened the description.

File: pkgs/development/python-modules/ipymarkup/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, intervaltree
+}: buildPythonPackage rec {
```
Comment (natsukium) [inline pkgs/development/python-modules/ipymarkup/default.nix:5]: nit
```suggestion
}:

buildPythonPackage rec {
```

File: pkgs/development/python-modules/ipymarkup/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  version = "0.9.0";
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-P0v6EP1mKTIBr4SEp+a8tyO/LjPZpqoAiCZxr5yiaRE=";
+  };
+
+  propagatedBuildInputs = [ intervaltree ];
+  pythonImportCheck = [ "ipymarkup" ];
```
Comment (natsukium) [inline pkgs/development/python-modules/ipymarkup/default.nix:16]:
```suggestion
  pythonImportCheck = [ "ipymarkup" ];
  
  # upstream has no tests
  doCheck = false;
```
Would you please add `doCheck = false`?

File: pkgs/development/python-modules/navec/default.nix [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-TyNHSxwnmvbGBfhOeHPofEfKWLDFOKP50w2QxgnJ/SE=";
+  };
+
+  propagatedBuildInputs = [ numpy razdel ];
+  nativeCheckInputs = [ pytestCheckHook gensim ];
+  disabledTests = [ "test_gensim" ];
```
Comment (natsukium) [inline pkgs/development/python-modules/navec/default.nix:20]: Could you add a comment about why we disabled it?

File: pkgs/development/python-modules/slovnet/default.nix [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-AtIle9ybnMHSQr007iyGHGSPcIPveJj+FGirzDge95k=";
+  };
+
+  propagatedBuildInputs = [ numpy navec razdel ];
+  nativeCheckInputs = [ pytestCheckHook ];
+  pytestFlagsArray = [ "tests/" ];
+  disabledTestPaths = [ "tests/test_api.py" ];
```
Comment (natsukium) [inline pkgs/development/python-modules/slovnet/default.nix:21]: ditto

Comment (natsukium) [reply pkgs/development/python-modules/slovnet/default.nix:21]: I missed this.
> Some tests disabled due to sandbox unfriendly tests scripts, e.g. slovnet trying to download model binary artifacts.
```suggestion
  disabledTestPaths = [ 
    # tries to download model binary artifacts
    "tests/test_api.py"
  ];
```

File: pkgs/development/python-modules/ipymarkup/default.nix (hunk 5 above)
Comment (npatsakula) [reply pkgs/development/python-modules/ipymarkup/default.nix]: Fixed for all commits in PR.

File: pkgs/development/python-modules/ipymarkup/default.nix (hunk 6 above)
Comment (npatsakula) [reply pkgs/development/python-modules/ipymarkup/default.nix:18]: I've added this change into `ipymarkup` commit.

File: pkgs/development/python-modules/navec/default.nix (hunk 7 above)
Comment (npatsakula) [reply pkgs/development/python-modules/navec/default.nix:23]: I've added little comment into `navec` commit. Thanks for mentioning it!

File: pkgs/development/python-modules/slovnet/default.nix (hunk 8 above)
Comment (npatsakula) [reply pkgs/development/python-modules/slovnet/default.nix]: I've added this comment into `slovnet` commit. Thanx!

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 253417` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages built:</summary>
  <ul>
    <li>python310Packages.ipymarkup</li>
    <li>python310Packages.ipymarkup.dist</li>
    <li>python310Packages.natasha</li>
    <li>python310Packages.natasha.dist</li>
    <li>python310Packages.navec</li>
    <li>python310Packages.navec.dist</li>
    <li>python310Packages.razdel</li>
    <li>python310Packages.razdel.dist</li>
    <li>python310Packages.slovnet</li>
    <li>python310Packages.slovnet.dist</li>
    <li>python310Packages.yagry</li>
    <li>python310Packages.yagry.dist</li>
    <li>python311Packages.ipymarkup</li>
    <li>python311Packages.ipymarkup.dist</li>
    <li>python311Packages.natasha</li>
    <li>python311Packages.natasha.dist</li>
    <li>python311Packages.navec</li>
    <li>python311Packages.navec.dist</li>
    <li>python311Packages.razdel</li>
    <li>python311Packages.razdel.dist</li>
    <li>python311Packages.slovnet</li>
    <li>python311Packages.slovnet.dist</li>
    <li>python311Packages.yagry</li>
    <li>python311Packages.yagry.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Pull request merged**